sky/server/server.py

```diff
@@ -1104,16 +1107,19 @@ async def complete_storage_name(incomplete: str,) -> List[str]:
         logger.info(f'Starting SkyPilot API server, workers={num_workers}')
         # We don't support reload for now, since it may cause leakage of request
         # workers or interrupt running requests.
-        uvicorn.run('sky.server.server:app',
-                    host=cmd_args.host,
-                    port=cmd_args.port,
-                    workers=num_workers)
+        config = uvicorn.Config('sky.server.server:app',
+                                host=cmd_args.host,
+                                port=cmd_args.port,
+                                workers=num_workers)
+        skyuvicorn.run(config)
     except Exception as exc:  # pylint: disable=broad-except
         logger.error(f'Failed to start SkyPilot API server: '
                      f'{common_utils.format_exception(exc, use_bracket=True)}')
         raise
     finally:
         logger.info('Shutting down SkyPilot API server...')
         for sub_proc in sub_procs:
-            sub_proc.terminate()
-            sub_proc.join()
+            # Ignore dead or not started processes.
+            if sub_proc.is_alive():
+                sub_proc.terminate()
+                sub_proc.join()
```

Comment:
Wondering if we need to clean up the dead processes.

Reply:
Not sure, `sub_proc.is_alive()` returns False when the process is not started or the process has exited with a returncode, so I think there is no need to cleanup the real process. Added `sub_proc.close()` for all processes to cleanup the process object for completeness and tested manually with SIGTERM a sub process and then exit the server.

Reply:
Good catch!